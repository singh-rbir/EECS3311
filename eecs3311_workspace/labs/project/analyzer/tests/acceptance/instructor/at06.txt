add_class("A")
add_class("B")

-- Error: Duplicated parameter names.

-- Users are expected to use INTEGER and BOOLEAN to specify the primitive types, not int and boolean.
add_command ("A", "ca", <<["x", "int"], ["y", "boolean"], ["z", "C"]>>)

-- Here the return type is non-existing (should be INTEGER instead),
-- but this error has lower priority (see the table in instructions).
add_query ("A", "ca", <<["x", "int"], ["y", "boolean"], ["z", "C"]>>, "int")
add_query ("A", "cq", <<["x", "int"], ["y", "boolean"], ["z", "C"]>>, "C")

-- Error: non-existing return type for attributes and queries
add_attribute ("A", "att", "C")
add_query ("A", "ca", <<["x", "INTEGER"], ["y", "BOOLEAN"], ["z", "A"]>>, "C")



