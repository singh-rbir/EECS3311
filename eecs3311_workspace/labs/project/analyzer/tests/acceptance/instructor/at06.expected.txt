  Status: OK.
  Number of classes being specified: 0
->add_class("A")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_class("B")
  Status: OK.
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_command("A","ca",<<["x", "int"], ["y", "boolean"], ["z", "C"]>>)
  Status: Error (Parameter types do not refer to primitive types or existing classes: int, boolean, C).
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_query("A","ca",<<["x", "int"], ["y", "boolean"], ["z", "C"]>>,"int")
  Status: Error (Parameter types do not refer to primitive types or existing classes: int, boolean, C).
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_query("A","cq",<<["x", "int"], ["y", "boolean"], ["z", "C"]>>,"C")
  Status: Error (Parameter types do not refer to primitive types or existing classes: int, boolean, C).
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_attribute("A","att","C")
  Status: Error (Return type does not refer to a primitive type or an existing class: C).
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_query("A","ca",<<["x", "INTEGER"], ["y", "BOOLEAN"], ["z", "A"]>>,"C")
  Status: Error (Return type does not refer to a primitive type or an existing class: C).
  Number of classes being specified: 2
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
    B
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
